src/main/scala/com/ibm/couchdb/api/builders/GetManyDocumentsQueryBuilder.scala

```diff
@@ -160,13 +205,69 @@ case class GetManyDocumentsQueryBuilder(
   }
 
   private def queryWithoutIds[Q: R](ps: Map[String, String]): Task[Q] = {
-    postQuery[Q](client, db, s"/$db/_all_docs", ps)
+    QueryUtils.postQuery[Q](client, db, s"/$db/_all_docs", ps)
   }
 
   private def queryByIds[Q: R](ids: Seq[String], ps: Map[String, String]): Task[Q] = {
     if (ids.isEmpty)
       Res.Error("not_found", "No IDs specified").toTask
     else
-      queryByIds[String, Q](client, db, s"/$db/_all_docs", ids, ps)
+      QueryUtils.queryByIds[String, Q](client, db, s"/$db/_all_docs", ids, ps)
+  }
+}
+
+object GetManyDocumentsQueryBuilder {
+
+  trait BuildReady[T] {
+    def build: T
+  }
+
+  def apply(client: Client, db: String, typeMapping: TypeMapping):
+  GetManyDocumentsQueryBuilder[ExcludeDocs, MissingNotAllowed, AnyDocType] =
+    new GetManyDocumentsQueryBuilder[ExcludeDocs, MissingNotAllowed, AnyDocType](
+      client, db, typeMapping)
+
+  implicit def buildBasic(
```

Comment:
After thinking about it more and considering that we want to have different return types based on the configuration, this approach looks nice. There is quite a bit of boilerplate, maybe some of it can be removed by adding implementations of BuildReady.